### DISCUSSION TITLE
No encryption method for xchacha20-ietf-poly1305

### DISCUSSION THREAD

**Discussion by hamzamalikdevgalixo** [OutlineFoundation/outline-sdk#336] [Q&A] [answered]
There is no support for `xchacha20-ietf-poly1305`. golang.org/x/crypto/ has support for this. Is there any update going on regarding this

**Comment by amircybersec** [comment]
@hamzamalikdevgalixo may I ask what server you are using with Outline client? As far as Outline server supports all client side (app) encryptions. Please let me know if you have any suggestions on code snippets that show how this can be added to the SDK.

**Reply by hamzamalikdevgalixo** [reply]
We are trying to integrate the SDK in our app. The server is configured for `xchacha20-ietf-poly1305` but the sdk does not have support for it.

**Reply by amircybersec** [reply]
@hamzamalikdevgalixo are you using xray or singbox on the server side?

**Reply by hamzamalikdevgalixo** [reply]
We are using `shadowsocks-libev` on the server side

**Reply by amircybersec** [reply]
@hamzamalikdevgalixo Looking at the spec, seems like these three crypto are part of the spec:

<img width="707" alt="Screenshot 2024-11-27 at 1 25 22 AM" src="https://github.com/user-attachments/assets/c535d802-43cf-4c6e-a642-5e43bcf83ab1">

https://shadowsocks.org/doc/aead.html

May I ask why not use one of the supported encryption methods? Is there any obvious advantage in using `xchacha20-ietf-poly1305` or is it just for maximum compatibility?

**Reply by hamzamalikdevgalixo** [reply]
Yes it has support for those but I'm looking for maximum compatibility. I don't want to change the infrastructure setup that's done and its working well in our other platform as well.

**Reply by amircybersec** [reply]
@hamzamalikdevgalixo it seems like other shadowsocks implementations are deprecating `xchacha20-ietf-poly1305`. It is not clear why (mostly for simplification, maintenance, and performance I guess) 

https://github.com/shadowsocks/shadowsocks-android/issues/2697

Maybe others can chime in and see if it is a good idea to add it to SDK or there is a good reason not to.

**Comment by fortuna** [comment, marked as answer]
xchacha20-ietf-poly1305 seems to be an abandoned IETF proposal: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-xchacha-03

We would like to stick to the standard crypto. Support for alternative ciphers may open up the code to vulnerabilities. I strongly recommend moving away from that cipher in favor for the IETF ones.

**Comment by mgma9614-cloud** [comment]
ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTpzeGdlaXNEcVBYdkhUYWxWbzJ1RW9U@159.89.196.236:33151/?outline=1

**Comment by mgma9614-cloud** [comment]
ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTpzeGdlaXNEcVBYdkhUYWxWbzJ1RW9U@159.89.196.236:33151/?outline=1
